### ISSUE TITLE
Null pointer exception running oai indexing on vteckworks-dev with DSpace 6

### ISSUE THREAD

**Issue by alawvt** [VTUL/vtechworks#623]
The crontab has been created on vtechworks-dev for DSpace 6. The oai indexing only produced ~11,000 items instead of ~75,000 so I ran:
```
alaw@vtechworks-dev:~$ sudo -u vtechworks /dspace/bin/dspace oai import -c > /dev/null
java.lang.NullPointerException
	at org.dspace.xoai.app.XOAI.willChangeStatus(XOAI.java:438)
	at org.dspace.xoai.app.XOAI.index(XOAI.java:368)
	at org.dspace.xoai.app.XOAI.index(XOAI.java:280)
	at org.dspace.xoai.app.XOAI.indexAll(XOAI.java:227)
	at org.dspace.xoai.app.XOAI.index(XOAI.java:134)
	at org.dspace.xoai.app.XOAI.main(XOAI.java:560)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.dspace.app.launcher.ScriptLauncher.runOneCommand(ScriptLauncher.java:229)
	at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:81)
```
So I tried `alaw@vtechworks-dev:~$ sudo -u vtechworks /dspace/bin/dspace oai import -c`, which got to:

```
20000 items imported so far...
20100 items imported so far...
java.lang.NullPointerException
java.lang.NullPointerException
	at org.dspace.xoai.app.XOAI.willChangeStatus(XOAI.java:438)
	at org.dspace.xoai.app.XOAI.index(XOAI.java:368)
	at org.dspace.xoai.app.XOAI.index(XOAI.java:280)
	at org.dspace.xoai.app.XOAI.indexAll(XOAI.java:227)
	at org.dspace.xoai.app.XOAI.index(XOAI.java:134)
	at org.dspace.xoai.app.XOAI.main(XOAI.java:560)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.dspace.app.launcher.ScriptLauncher.runOneCommand(ScriptLauncher.java:229)
	at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:81)
```
Ideas?

**Comment by keithgee** [comment]
Not sure. Does this work on the other servers?

**Comment by alawvt** [comment]
The cron job on all three servers is `/dspace/bin/dspace oai import -o > /dev/null`. It seems to be working ok on prod because the record total, `https://vtechworks.lib.vt.edu/oai/request?verb=ListRecords&metadataPrefix=oai_dc`, 75849, is close to the actual number of items. On dev, it is only https://vtechworks-dev.cloud.lib.vt.edu/oai/request?verb=ListRecords&metadataPrefix=oai_dc, 20160.

**Comment by keithgee** [comment]
This appears to be a known issue with DSpace 6.3:
https://jira.duraspace.org/browse/DS-3969

Edit: This issue report references XMLWorkflow, but it appears to be the same or similar problem.

**Comment by keithgee** [comment]
I will test a fix for this on my LDE, and if successful, apply it to vtechworksdev.

**Comment by alawvt** [comment]
Regarding [DS-3969](https://jira.duraspace.org/browse/DS-3969), we don't have XMLWorkflow enabled, so that case does not affect us.

There is a more extensive discussion of this issue and possible fixes in http://dspace.2283337.n4.nabble.com/DSpace-6-3-OAI-indexing-is-failing-td4688714.html.

**Comment by keithgee** [comment]
Christian's fix is essentially what I tested on my LDE yesterday. Today I've pulled his commits into the [fix_oai_indexer_npe_dspace6_3](https://github.com/VTUL/vtechworks/tree/fix_oai_indexer_npe_dspace6_3) branch and tested on the cloud server. Indexing completed successfully.

**Comment by alawvt** [comment]
Resolved with #624. Thank you @keithgee.
